src/components/Popup/Popup.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import CSSProperties = React.CSSProperties
+
+import { childrenExist, customPropTypes, isBrowser } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import { ItemShorthand, Extendable } from '../../../types/utils'
+import Portal from '../Portal'
+import PopupContent from './PopupContent'
+
+type PopupPosition =
+  | 'top start'
+  | 'top end'
+  | 'bottom start'
+  | 'bottom end'
+  | 'start center'
+  | 'end center'
+  | 'top center'
+  | 'bottom center'
+
+const POSITIONS: PopupPosition[] = [
+  'top start',
+  'top end',
+  'bottom start',
+  'bottom end',
+  'start center',
+  'end center',
+  'top center',
+  'bottom center',
+]
+
+export interface IPopupProps {
+  as?: any
+  basic?: boolean
+  className?: string
+  content?: ItemShorthand | ItemShorthand[]
+  position?: PopupPosition
+  style?: CSSProperties
+  trigger: JSX.Element
+}
+
+export interface IPopupState {
+  style?: CSSProperties
+}
+
+/**
+ * A Popup displays additional information on top of a page.
+ * @accessibility This is example usage of the accessibility tag.
+ * This should be replaced with the actual description after the PR is merged
+ */
+export default class Popup extends React.Component<Extendable<IPopupProps>, IPopupState> {
+  private popupOffset = 8
+  private triggerRef: HTMLElement
+  private popupCoords: ClientRect | DOMRect | undefined
+
+  public static Content = PopupContent
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Basic CSS styling for the popup */
+    basic: PropTypes.bool,
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Simple text content for the popover. */
+    content: customPropTypes.itemShorthand,
+
+    /** Position for the popup. */
+    position: PropTypes.oneOf(POSITIONS),
+
+    /** Element to be rendered in-place where the popup is defined. */
+    trigger: PropTypes.node.isRequired,
+
+    /** Custom style to be applied for component. */
+    style: PropTypes.object,
+  }
+
+  public static defaultProps = {
+    position: 'top start',
+  }
+
+  public state = { style: {} }
+
+  public render() {
+    const { basic, children, content, trigger } = this.props
+    const { style } = this.state
+
+    return (
+      <Portal onMount={this.handlePortalMount} trigger={trigger} triggerRef={this.handleTriggerRef}>
+        <Popup.Content basic={basic} triggerRef={this.handlePopupRef} styles={{ root: style }}>
```

Comment:
```triggerRef``` will be actually set to popup content Ref, so a bit confused about the naming
@Bugaa92

Reply:
changed to `innerRef`, thx